nav2_bringup/README.md

```diff
@@ -45,12 +45,25 @@ ros2 launch turtlebot3_bringup turtlebot3_state_publisher.launch.py use_sim_time
 
 ### Terminal 3: Launch Nav2
```

Comment:
At the top of this readme adding that this is a very flexible example for nav2 bring up that can be modified for different maps/robots/hardware/worlds/etc. It is our expectation for an application specific thing you're mirroring nav2_bringup package and modifying it for your sp. maps/robots/bringup needs. So this is an applied and working demonstration for the default system bringup with many options that can be easily modified. 

Usual robot stacks will have a `<robot_name>_nav` package with config/bringup files and this is that for the general case to base a specific robot system off of

Reply:
Also, some discussion about why we do manual composition rather than dynamic composition in the launch file due to various issues in place (parameters, lifecycle components not playing well, etc). Eventually we'd do dynamic components but for now, manually composed. Link the relevent tickets / issues so that if they get resolved someone can take note and potentially enable dynamic composition